pinot-core/src/main/java/com/linkedin/pinot/core/query/aggregation/groupby/AggregationGroupByTrimmingService.java

```diff
@@ -171,9 +177,16 @@ public AggregationGroupByTrimmingService(@Nonnull AggregationFunction[] aggregat
       for (Map.Entry<String, Comparable> entry : finalResultMap.entrySet()) {
         String groupKey = entry.getKey();
         Comparable finalResult = entry.getValue();
-        priorityQueue.add(new GroupKeyResultPair(groupKey, finalResult));
-        if (priorityQueue.size() > _groupByTopN) {
-          priorityQueue.poll();
+
+        GroupKeyResultPair newValue = new GroupKeyResultPair(groupKey, finalResult);
+        if (priorityQueue.size() >= _groupByTopN) {
```

Comment:
This code looks the same as the code above. We can probably take out these as a separate function to reuse the code. i.e. `addToPriorityQueue(priorityQueue, maxSize)`